### PR TITLE
initial version of quick start multi-cluster scripts

### PR DESCRIPTION
**Pull request by ayj**

### CODE REVIEW

File: samples/multicluster/setup-mesh.sh [hunk 1]
```diff
@@ -0,0 +1,432 @@
+#!/bin/bash
+
+# Copyright 2018 Istio Authors
```
Comment (geeknoid) [inline samples/multicluster/setup-mesh.sh]: Remove year

File: samples/multicluster/setup-mesh.sh [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+#
```
Comment (geeknoid) [inline samples/multicluster/setup-mesh.sh]: I'm uncomfortable with mixing bookinfo into this. I'd like to have a clean "setup for multicluster" story, independent of the bookinfo demo. Could we just have two scripts instead?

File: samples/multicluster/setup-mesh.sh (hunk 1 above)
Comment (ayj) [reply samples/multicluster/setup-mesh.sh]: done

File: samples/multicluster/setup-mesh.sh (hunk 2 above)
Comment (ayj) [reply samples/multicluster/setup-mesh.sh]: done

**Comment by ayj** [comment]
I'll be the first to admit this shouldn't be all bash. We could re-write in bash, or do things properly in master for 1.5.

File: samples/multicluster/setup-mesh.sh [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+# Organization name to use when creating the root and intermediate certificates.
+: "${ORG_NAME:?ORG_NAME not set}"
+
+#
+# derived configuration
+#
+
+# Resolve to an absolute path
+WORKDIR=$(readlink -f "${WORKDIR}")
```
Comment (geeknoid) [inline samples/multicluster/setup-mesh.sh]: readlink -f doesn't work on macOS...

~/go/src/istio.io/istio.io/mydir$ ./setup-mesh.sh create-mesh
readlink: illegal option -- f
usage: readlink [-n] [file ...]

File: samples/multicluster/setup-mesh.sh [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+  cleanup() {
+    rm -rf "${TMP_KUBECONFIG_DIR}"
+  }
+  trap cleanup EXIT
+
+  MESH_ID=$(grep -e "^mesh: \(.*\)$" "${CLUSTER_LIST}" |cut -d: -f2 | sed 's/ //')
+
+  # track existing contexts that were already merged. A random suffix
+  # is added to non-unique contexts.
+  declare -A EXISTING_CONTEXTS
```
Comment (geeknoid) [inline samples/multicluster/setup-mesh.sh]: -A doesn't work on macOS.

File: samples/multicluster/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+```bash
+./setup-mesh.sh add-cluster ${KUBECONFIG0} ${CONTEXT0} ${NETWORK1}
+./setup-mesh.sh add-cluster ${KUBECONFIG1} ${CONTEXT1} ${NETWORK1}
+./setup-mesh.sh apply
+```
+
+Install the sample bookinfo application in each cluster.
+
+```bash
+./setup-bookinfo.sh install
```
Comment (geeknoid) [inline samples/multicluster/README.md]: Can you add this script to the PR?

File: samples/multicluster/setup-mesh.sh [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+
+check_prerequisties() {
+  if [[ ! -d "${WORKDIR}" ]]; then
+    echo "error: workspace directory '${WORKDIR}' not found. Run 'mkdir workspace && export WORKDIR=\$(pwd)/workspace' "
+    exit 1
+  fi
+}
+
+usage() {
+  echo "Usage: $0 create-mesh | add-cluster | remove-cluster | apply | teardown
```
Comment (geeknoid) [inline samples/multicluster/setup-mesh.sh]:
```suggestion
  echo "Usage: $0 create-mesh | apply | teardown
```

File: samples/multicluster/setup-mesh.sh [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+
+create-mesh
```
Comment (geeknoid) [inline samples/multicluster/setup-mesh.sh]: Should this be called something else since it's not really creating a mesh? How about just "prep"?

Comment (ayj) [reply samples/multicluster/setup-mesh.sh]: done

File: samples/multicluster/README.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+```bash
+export WORKDIR=mesh-workspace
+[ ! -d "${WORKDIR}" ] && mkdir ${WORKDIR}
+```
+
+Prepare the initial configuration for the mesh. This creates a root key and cert
+that will sign intermediate certs for each cluster.
+
+```bash
+./setup-mesh.sh create-mesh
```
Comment (geeknoid) [inline samples/multicluster/README.md]:
```suggestion
./setup-mesh.sh prep
```

**Pull request merged**